Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1687,7 +1695,46 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
                 allReportedFileAccesses: allFileAccesses,
                 detouringStatuses: result.DetouringStatuses,
                 maxDetoursHeapSize: maxDetoursHeapSize,
-                containerConfiguration: m_containerConfiguration);
+                containerConfiguration: m_containerConfiguration,
+                pipProperties: pipProperties);
+        }
+
+        private async Task<Dictionary<string, int>> UpdatePipPropertiesCountsAsync(SandboxedProcessResult result, Dictionary<string, int> pipProperties)
+        {
+            const string PipPropertyPrefix = "PipProperty_";
+
+            Regex pipPropertyRegex = new Regex(PipPropertyPrefix + @"(?<" + PipPropertyPrefix + @">.+?)_EndProperty", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.Multiline);

```

Comment:
Please add a brief comment what this Regex is trying to achieve. A comment accompanied with an example would be better -- adding unit tests would be great.